gquantlab/gquantlab/handlers.py

```diff
@@ -0,0 +1,55 @@
+import json
+
+from notebook.base.handlers import APIHandler
+from notebook.utils import url_path_join
+import tornado
+from gquant.dataframe_flow import TaskGraph
+from .server_utils import (get_nodes, add_nodes)
+import os
+
+class RouteHandlerLoadGraph(APIHandler):
+    @tornado.web.authenticated
+    def get(self):
+        self.finish("abcde")
+
+    @tornado.web.authenticated
+    def post(self):
+        # input_data is a dictionnary with a key "name"
+        input_data = self.get_json_body()
+        task_graph = TaskGraph(input_data)
+        nodes_and_edges = get_nodes(task_graph)
+        self.finish(json.dumps(nodes_and_edges))
+
+
+class RouteHandlerLoadGraphFromPath(APIHandler):
+
+    @tornado.web.authenticated
+    def post(self):
+        # input_data is a dictionnary with a key "name"
+        input_data = self.get_json_body()
+        task_graph = TaskGraph.load_taskgraph(input_data['path'])
+        nodes_and_edges = get_nodes(task_graph)
+        self.finish(json.dumps(nodes_and_edges))
+
+
+class RouteHandlerLoadAllNodes(APIHandler):
+
+    @tornado.web.authenticated
+    def get(self):
+        # input_data is a dictionnary with a key "name"
+        result = add_nodes()
+        self.finish(json.dumps(result))
+
+
+def setup_handlers(web_app):
+    host_pattern = ".*$"
+    base_url = web_app.settings["base_url"]
+    os.environ['GQUANTROOT'] = os.getcwd()
```

Comment:
You are right. `os.getcwd()` is just the default behavior

Reply:
I make a close look at the GQUANTROOT variable. It can only be os.getcwd() as it is used to send the gquantlab working directory information to some utility functions to help to find the path of the files in the disk